docarray/array/array_stacked.py

```diff
@@ -91,10 +91,30 @@ def _from_columns(
         da_stacked._docs = docs
         return da_stacked
 
+    def to(self: T, device: str):
+        """Move all tensors of this DocumentArrayStacked to the given device
+
+        :param device: the device to move the data to
+        """
+        for field in self._columns.keys():
+            col = self._columns[field]
+            if isinstance(col, AbstractTensor):
+                # the casting below is arbitrary, in reality `col` could be of any
+                # subclass of AbstractTensor. But to make mypy happy we have to cast
+                # it to a concrete subclass thereof
+                # see mypy issue: https://github.com/python/mypy/issues/14421
+                col_ = cast('TorchTensor', col)
+                self._columns[field] = col_.get_comp_backend().to_device(col_, device)
+            elif isinstance(col, NdArray):
```

Comment:
I think this is better than the many `if` statements because it does not require any changes once we add more ML frameworks (and abstract tensors).
Also, imo, if mypy cannot grasp something for whatever reason, and that requires us to do something a bit whacky, then I would rather keep the whakiness in the typing machinery, since it does not affect the actual code that runs.

Reply:
Ah in that case we dont even need the elif. The elif never gets used anyways